pkg/generators/builder/builder.go

```diff
@@ -100,27 +102,75 @@ func (b *BuilderPatternGenerator) Init(c *generator.Context, w io.Writer) error
 }
 
 func (b *BuilderPatternGenerator) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
-	log.Debugf("Checking type: %s", t.Name.Name)
-	if !b.needsGeneration(t) {
-		return nil
-	}
 	log.Infof("Generating type: %s", t.Name.Name)
 
 	sw := generator.NewSnippetWriter(w, c, "$", "$")
 
-	sw.Do("//auto generated\n", nil) // TODO can be removed once setters are constructed for types
+	// generate constructor
+	if hasObjectMetaEmbedded(t) {
+		parentTypeOfObjectMeta := getParentOfEmbeddedType(t, ObjectMeta)
+		b.imports.AddType(parentTypeOfObjectMeta)
+		b.imports.AddType(getMemberTypeFromType(parentTypeOfObjectMeta, ObjectMeta))
+		sw.Do(snippets.GenerateConstructorForObjectMeta(t))
+	} else {
+		sw.Do(snippets.GenerateEmptyConstructor(t, true))
+	}
+
 	if hasTypeMetaEmbedded(t) {
-		parentTypeOfTypeMeta := getParentOfTypeMeta(t)
+		parentTypeOfTypeMeta := getParentOfEmbeddedType(t, TypeMeta)
 		b.imports.AddType(parentTypeOfTypeMeta)
-		b.imports.AddType(getTypeMetaFromType(parentTypeOfTypeMeta))
+		b.imports.AddType(getMemberTypeFromType(parentTypeOfTypeMeta, TypeMeta))
 		sw.Do(snippets.GenerateDeepCopy(t))
 	}
 
-	// TODO generate setters for struct
+	// generate setters for struct
+	if hasObjectMetaEmbedded(t) {
+		parentTypeOfObjectMeta := getParentOfEmbeddedType(t, ObjectMeta)
+		objectMetaType := getMemberTypeFromType(parentTypeOfObjectMeta, ObjectMeta)
+		b.generateSettersForType(sw, t, objectMetaType)
+	}
+	for _, member := range t.Members {
+		b.generateSettersForType(sw, t, member.Type)
+	}
 
 	return sw.Error()
 }
 
+func (b *BuilderPatternGenerator) generateSettersForType(sw *generator.SnippetWriter, root *types.Type, parent *types.Type) {
+	setter := snippets.NewSetter(root, parent, true)
+
+	for _, m := range parent.Members {
+		if m.Embedded || !includeMember(parent, m) {
+			continue
+		}
+
+		if m.Type.Kind == types.Map {
+			sw.Do(setter.GenerateSetterForMap(m))
+		} else if m.Type.Kind == types.Slice {
+			sliceType := m.Type.Elem
+
+			switch sliceType.Kind {
+			case types.Struct:
+				// skip adding setters for un-enabled structs
+				if inputType := b.getTypeEnabledForGeneration(sliceType); inputType != nil {
+					sw.Do(setter.GenerateSetterForEmbeddedSlice(m, inputType))
+				}
+			default:
+				sw.Do(setter.GenerateSetterForMemberSlice(m))
+			}
+		} else if m.Type.Kind == types.Struct {
+			// skip adding setters for un-enabled structs
+			if inputType := b.getTypeEnabledForGeneration(m.Type); inputType != nil {
+				sw.Do(setter.GenerateSetterForEmbeddedStruct(m, inputType))
+			}
+		} else if m.Type.Kind == types.Pointer && m.Type.Elem.Kind == types.Builtin {
+			sw.Do(setter.GenerateSetterForPointerToBuiltinType(m))
+		} else {
+			sw.Do(setter.GenerateSetterForPrimitiveType(m))
+		}
```

Comment:
Yeah I was hoping to use switch case, but the check on L166 on two variables makes it hard. Since the else case of that needs to fallthrough to L168. I'll take a look